sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/encoders/ExpressionEncoder.scala

```diff
@@ -80,7 +80,11 @@ object ExpressionEncoder {
    * name/positional binding is preserved.
    */
   def tuple(encoders: Seq[ExpressionEncoder[_]]): ExpressionEncoder[_] = {
-    // TODO: check if encoders length is more than 22 and throw exception for it.
+    if (encoders.length > 22) {
```

Comment:
`ExpressionEncode.tuple` has a chance to receive more than 22 elements in `encoders`? It seems the catalyst package is internal only and all the caller of this `tuple` passes [at most 5 elements](https://github.com/apache/spark/blob/29d902124581b2163e49baa6d379025a6cf1fa1a/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/encoders/ExpressionEncoder.scala#L160). If so, `assert` here?

Reply:
Ah yes if we tend to see entire catalyst package to internal API, Encoders doesn't expose adding such elements. In that case, `assert` also sounds good to me. @dongjoon-hyun WDYT?

Reply:
@maropu . There is `Dataset.selectUntyped` using `ExpressionEncoder.tuple(encoders)`. That is also internal helper function which uses with at most 5 elements, but some users can extend `Dataset` and use it with more parameters. BTW, it's true that Apache Spark doesn't guarantee anything about that. I prefer the current one, but @maropu 's suggestion also look good.

Reply:
To me, no strong preference given it would bring similar error message, though it would throw different exceptions (even both two different exceptions make sense for given point, though I think `UnsupportedOperationException` is more clearer to represent what's happening).

Change to `require` looks to help reducing lines and simplifies the code. Couldn't find other difference. @maropu Are you OK with stick to current one? Or do you have strong preference to use `require` on this?

Reply:
Its ok as it is.

Reply:
Cool, thanks for understanding, @maropu ! cc. @dongjoon-hyun